specification/cognitiveservices/data-plane/QnAMaker/stable/v4.0/QnAMaker.json

```diff
@@ -0,0 +1,1275 @@
+        ],
+        "responses": {
+          "200": {
+            "description": "Details of the endpoint keys generated.",
+            "schema": {
+              "$ref": "#/definitions/EndpointKeysDTO"
+            },
+            "x-nullable": true
+          },
+          "default": {
+            "description": "Error response.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Successful query": {
+            "$ref": "./examples/SuccessfulSetEpKeys.json"
+          }
+        }
+      }
+    },
+    "/alterations": {
+      "get": {
+        "tags": [
+          "Alterations"
+        ],
+        "summary": "Download alterations from runtime.",
+        "operationId": "Download Alterations",
+        "parameters": [],
+        "consumes": [
+          "application/json"
+        ],
+        "produces": [
+          "application/json"
+        ],
+        "responses": {
+          "200": {
+            "description": "Alterations data.",
+            "schema": {
+              "$ref": "#/definitions/WordAlterationsDTO"
+            },
+            "x-nullable": true
+          },
+          "default": {
+            "description": "Error response.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Successful query": {
+            "$ref": "./examples/SuccessfulGetAlts.json"
+          }
+        }
+      },
+      "put": {
+        "tags": [
+          "Alterations"
+        ],
+        "summary": "Replace alterations data.",
+        "operationId": "Replace Alterations",
+        "parameters": [
+          {
+            "$ref": "#/parameters/WordAlterations"
+          }
+        ],
+        "consumes": [
+          "application/json"
+        ],
+        "produces": [
+          "application/json"
+        ],
+        "responses": {
+          "204": {
+            "description": "HTTP 204 No Content."
+          },
+          "default": {
+            "description": "Error response.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Successful query": {
+            "$ref": "./examples/SuccessfulSetAlts.json"
+          }
+        }
+      }
+    },
+    "/knowledgebases": {
+      "get": {
+        "tags": [
+          "Knowledgebases"
+        ],
+        "summary": "Gets all knowledgebases for a user.",
+        "operationId": "Get Knowledgebases for user",
+        "parameters": [],
+        "consumes": [
+          "application/json"
+        ],
+        "produces": [
+          "application/json"
+        ],
+        "responses": {
+          "200": {
+            "description": "Collection of knowlegebases.",
+            "schema": {
+              "$ref": "#/definitions/KnowledgebasesDTO"
```

Comment:
for these "get all"/ "list" operations, have you looked at x-ms-pageable? https://github.com/Azure/autorest/tree/master/docs/extensions#x-ms-pageable 
I'm asking in case you don't know about it and your model/service want to support it that way, doesn't mean you need to change it here.

Reply:
Thanks we dont support paging anywhere